src/components/CheckEditor/CheckEditor.test.tsx

```diff
@@ -0,0 +1,344 @@
+import React from 'react';
+import { render, screen, waitFor, act, within } from '@testing-library/react';
+import { Check, IpVersion, CheckType, DnsResponseCodes, ResponseMatchType } from 'types';
+import { CheckEditor } from './CheckEditor';
+import { getInstanceMock } from '../../datasource/__mocks__/DataSource';
+import userEvent from '@testing-library/user-event';
+import { InstanceContext } from 'components/InstanceContext';
+jest.setTimeout(20000);
+
+const defaultCheck = {
+  job: '',
+  target: '',
+  frequency: 60000,
+  timeout: 3000,
+  enabled: true,
+  labels: [],
+  probes: [],
+  settings: {
+    ping: {
+      ipVersion: IpVersion.V4,
+      dontFragment: false,
+    },
+  },
+} as Check;
+
+const getMinimumCheck = (overrides: any = {}) => ({
```

Comment:
here comes the types police: `overrides: Partial<Check>` perhaps?

Reply:
⌨️ 🚔